File: solutions/c/grains/3/grains.c
#include "grains.h"

uint64_t square(uint8_t index) {
    return (index < 1 || index > 64) ? 0 : 1ull << (index - 1);
}

uint64_t total(void) {
    return (uint64_t)-1;
}

File: solutions/go/grains/1/grains.go
package grains

import "fmt"

func Square(number int) (uint64, error) {
    if number <= 0 || number > 64 {
        return 0, fmt.Errorf("error")
    }
	return uint64(1) << (number - 1), nil
}

func Total() uint64 {
	var sum uint64

    for i := range 64 + 1 {
        tmp, _ := Square(i)
        sum += tmp
    }

    return sum
}
